src/handlers/issue/generate-nft-mint-signature.ts

```diff
@@ -0,0 +1,127 @@
+import { BigNumber, BigNumberish, ethers, utils } from "ethers";
+import { getPayoutConfigByNetworkId } from "../../helpers/payout";
+import { MaxUint256 } from "@uniswap/permit2-sdk";
+import { keccak256, toUtf8Bytes } from "ethers/lib/utils";
+
+const NFT_MINTER_PRIVATE_KEY = process.env.NFT_MINTER_PRIVATE_KEY as string;
+const NFT_CONTRACT_ADDRESS = process.env.NFT_CONTRACT_ADDRESS as string;
+const NFT_CONTRACT_CHAIN_ID = parseInt(process.env.NFT_CONTRACT_CHAIN_ID as string);
```

Comment:
I think we should not rely on environment variables. Instead these should be read from the config yml.

Reply:
This is a contract we deploy and not something partners should change so I don't think it makes sense to put it in the config.

Reply:
Perhaps we should hard code it then and be sure to use `CREATE2` to specify the same deploy address on all networks that Permit2 is deployed on? @rndquu

Reply:
> This is a contract we deploy and not something partners should change so I don't think it makes sense to put it in the config.

Yes, we can either hardcode it either use env variables. I would use env variables right now and when https://github.com/ubiquity/nft-rewards/issues/2 is ready we could hardcode the address.

Reply:
@whilefoo please hard code. I just made a note about this in https://github.com/ubiquity/nft-rewards/issues/2.

Reply:
But if hardcode the private key anyone who can see the code can see the private key

Reply:
I am really confused. It doesn't help that I'm on my phone now and can't conveniently review the code in its entirety but why would a smart contract have a private key associated? Can you explain with full context?

Reply:
When you deploy the NFT contract you specify a minter's address. You need that minter's private key to sign mint requests. A bounty hunter can mint an NFT using that signature and pays the gas fee instead of us.

It's similar to Permit2 but in that case, the partner's wallet has the tokens so they specify their wallet's private key in `evmPrivateKeyEncrypted` field in the config.